### ISSUE TITLE
Lint/UnusedMethodArgument should be able to skip empty methods

### ISSUE THREAD

**Issue by zenspider** [bbatsov/rubocop#1601]
When defining the interface with an abstract class, you often define methods with args but with an empty body, eg:

```ruby
    ##
    # Record a result and output the Runnable#result_code. Stores the
    # result of the run if the run did not pass.

    def record result
    end
```

The doco describes what the argument is for. If the body is empty, assume it is an abstract method and don't lint on this arg.

**Comment by bbatsov** [comment]
Makes sense. We can introduce a config option for this.

**Comment by agrimm** [comment]
This problem also occurs with empty blocks, especially in auto-generated configuration scripts that have lots of commented out option setting things you can do.

**Comment by alexdowad** [comment]
So you would prefer that this would be enabled through config, rather than being the default?

I find warnings about "unused parameters" in a empty method just add noise.

**Comment by bbatsov** [comment]
Well, empty or not, they are unused and sometimes this might be accidental instead of intended - e.g. you got distracted and forgot to write the body of the method. So, yes - I'm fine with supporting this, but it should be configurable. Frankly, I think this is shortcoming of Ruby's design - probably it would have been better if abstract methods could be denoted by some manner. 

As for empty blocks - it seems to me that an empty block usually doesn't make much sense, right?

**Comment by alexdowad** [comment]
Very well. I can write up a PR. Any idea what the config parameter for `Lint/UnusedMethodArgument` should be called?

**Comment by bbatsov** [comment]
`IgnoreEmptyMethods` and `IgnoreEmptyBlocks`.

**Comment by alexdowad** [comment]
You got it.

**Comment by alexdowad** [comment]
Meaning, "OK, that is fine, you will have it very soon".

**Comment by bbatsov** [comment]
10x :-)
